tests/test-resize.lua

```diff
@@ -0,0 +1,412 @@
+local test_client = require("_client")
+local placement = require("awful.placement")
+local amouse = require("awful.mouse")
+
+local steps = {}
+
+table.insert(steps, function(count)
+    if count == 1 then  -- Setup.
+        test_client("foobar", "foobar")
+    elseif #client.get() > 0 then
+
+        client.get()[1] : geometry {
+            x      = 200,
+            y      = 200,
+            width  = 300,
+            height = 300,
+        }
+
+        return true
+    end
+end)
+
+table.insert(steps, function()
+    root.fake_input("button_press",1)
```

Comment:
Why this button press? I think it's so that `client.resize` works (it wants a button to be down). Could you add a comment explaining this (and perhaps move it down a bit to right before the call to `mouse.coords`)?